src/core/agent/services/identifierService.ts

```diff
@@ -303,6 +315,144 @@ class IdentifierService extends AgentService {
       );
     }
   }
+
+  async createMultisig(
+    ourIdentifier: string,
+    otherIdentifierContacts: GenericRecord[],
+    meta: Pick<
+      IdentifierMetadataRecordProps,
+      "displayName" | "colors" | "theme"
+    >
+  ): Promise<string | undefined> {
+    const ourMetadata = await this.getMetadataById(ourIdentifier);
+    this.validIdentifierMetadata(ourMetadata);
+    const ourAid = (await this.agent.modules.signify.getIdentifierByName(
+      ourMetadata.signifyName as string
+    )) as Aid;
+    const otherAids = [];
+    for (const contact of otherIdentifierContacts) {
+      const aid = await this.agent.modules.signify.resolveOobi(
+        contact.content.oobi as string
+      );
+      otherAids.push({
+        state: aid.response,
+      } as Pick<Aid, "state">);
+    }
+    const result = await this.agent.modules.signify.createMultisig(
+      ourAid,
+      otherAids
+    );
+    const multisigId = result.op.name.split(".")[1];
+    await this.createIdentifierMetadataRecord({
+      id: multisigId,
+      displayName: meta.displayName,
+      method: IdentifierType.KERI,
+      colors: meta.colors,
+      theme: meta.theme,
+      signifyName: utils.uuid(),
+      signifyOpName: result.op.name, //we save the signifyOpName here to sync the multisig's status later
+      isPending: result.op.done ? false : true, //this will be updated once the operation is done
+    });
+    return multisigId;
+  }
+  async hasJoinedMultisig(msgSaid: string): Promise<boolean> {
+    const notifications: MultiSigIcpNotification[] =
+      await this.agent.modules.signify.getNotificationsBySaid(msgSaid);
+    if (!notifications.length) {
+      return false;
+    }
+    const exn = notifications[0].exn;
+    const multisigId = exn.a.gid;
+    try {
+      const multiSig = await this.agent.modules.signify.getIdentifierById(
+        multisigId
+      );
+      if (multiSig) {
+        return true;
+      }
+    } catch (e) {
+      return false;
+    }
+    return false;
+  }
+
+  async joinMultisig(
+    notification: KeriNotification,
+    meta: Pick<
+      IdentifierMetadataRecordProps,
+      "displayName" | "colors" | "theme"
+    >
+  ): Promise<string | undefined> {
+    const msgSaid = notification.a.d as string;
+    const hasJoined = await this.hasJoinedMultisig(msgSaid);
+    if (hasJoined) {
+      await this.agent.genericRecords.deleteById(notification.id as string);
+      return;
+    }
+    const notifications: MultiSigIcpNotification[] =
+      await this.agent.modules.signify.getNotificationsBySaid(msgSaid);
+
+    if (!notifications.length) {
+      throw new Error(
+        `${IdentifierService.SAID_NOTIFICATIONS_NOT_FOUND} ${msgSaid}`
+      );
+    }
+    const exn = notifications[0].exn;
+    const rstate = exn.a.rstates;
+    const identifiers = await this.getIdentifiers();
+    const identifier = identifiers.find((identifier) => {
```

Comment:
We can cover this in a follow up story, but we should see what happens if a wallet controls 2 of the multi-sig identifiers (this would make no sense, but just in case)

Does `joinMultisig` get called twice for 2 notifications? And I think both of the calls will end up with the same identifier since `.find` will find the first occurrance

Reply:
Just tested this case. It will only create 1 multisig identifier.